### ISSUE TITLE
`array-prefer-some` false negative for `notArray.find()`

### ISSUE THREAD

**Issue by Hoishin** [sindresorhus/eslint-plugin-unicorn#1016]
`array-prefer-some` reports error for any `.find()` calls even if it's not array's method

```js
class Foo {
  find() {
    return "test"
  }
}

if (new Foo().find(t => t === "t")) {
  console.log("this find is reported")
}
```

Apologies if it was the case where it's just impossible without TypeScript type info

**Comment by sindresorhus** [comment]
Without TS type info, it's impossible to avoid false-positives. In this specific case, we could not report if the caller is a constructor call. Is the above example representable of the actual code?

**Comment by Hoishin** [comment]
That makes sense.

The code example I made it as simple as possible but this is closer to my actual code:

```ts
class A {
  // ...
  find (key: string) {
    const data = this.getItem(key)
    return typeof data !== "undefined" && data !== null
  }
  // ...
}

// ...

class B {
  store = new A()
  
  foo() {
    if (this.store.find(SOME_KEY)) {
      // ...
    }
  }
}
```

Now that I looked back, this particular code can be improved by renaming the defined `.find` to `.some` or something following ES naming concept.

**Comment by TkDodo** [comment]
we stumbled upon the same issue today, using a `QueryCache` from `react-query`, which has a `find` method, but no `some` method:

```
queryClient.getQueryCache().find(['foo']) ? 'foo' : 'bar'
```

**Comment by gigobyte** [comment]
I just encountered the issue above.

Ideally, the rule would also use type information to figure out if there is a `some` method. A configuration of whitelisted variable patterns that don't report an error would also be a good solution (e.g. `queryCache.find`)
